core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -382,6 +387,65 @@ protected static String getRandomRewriteMethod() {
     }
 
     protected String getRandomType() {
-        return (currentTypes.length == 0) ? MetaData.ALL : randomFrom(currentTypes);  
+        return (currentTypes.length == 0) ? MetaData.ALL : randomFrom(currentTypes);
+    }
+
+    /**
+     * Helper method to return a random field (mapped or unmapped) and a value
+     */
+    protected static Tuple<String, Object> getRandomFieldNameAndValue() {
+        // if no type is set then return random field name and value
+        if (currentTypes == null || currentTypes.length == 0) {
+            return new Tuple(randomAsciiOfLengthBetween(1, 10), randomAsciiOfLengthBetween(1, 50));
+        }
+        // mapped fields
+        String fieldName = randomFrom(mappedFieldNames);
+        Object value = randomAsciiOfLengthBetween(1, 50);
+        switch(fieldName) {
+            case STRING_FIELD_NAME:
+                value = rarely() ? randomUnicodeOfLength(10) : value; // unicode in 10% cases
+                break;
+            case INT_FIELD_NAME:
+                value = randomIntBetween(0, 10);
+                break;
+            case DOUBLE_FIELD_NAME:
+                value = randomDouble() * 10;
+                break;
+            case BOOLEAN_FIELD_NAME:
+                value = randomBoolean();
+                break;
+            case DATE_FIELD_NAME:
+                value = new DateTime(System.currentTimeMillis(), DateTimeZone.UTC).toString();
+                break;
+        } // all other fields assigned to random string
+
+        // unmapped fields
+        if (randomBoolean()) {
+            fieldName = randomAsciiOfLengthBetween(1, 10);
+        }
+        return new Tuple(fieldName, value);
+    }
+    
+    protected static Fuzziness randomFuzziness(String fieldName) {
+        Fuzziness fuzziness = Fuzziness.AUTO;
+        switch (fieldName) {
+            case INT_FIELD_NAME:
+                fuzziness = Fuzziness.build(randomIntBetween(3, 100));
+                break;
+            case DOUBLE_FIELD_NAME:
+                fuzziness = Fuzziness.build(1 + randomFloat() * 10);
+                break;
+            case DATE_FIELD_NAME:
+                fuzziness = Fuzziness.build(randomTimeValue());
+                break;
+        }
+        if (randomBoolean()) {
+            fuzziness = Fuzziness.fromEdits(randomIntBetween(0, 2));
+        }
+        return fuzziness;
+    }
+
+    protected static boolean isNumericFieldName(String fieldName) {
```

Comment:
to be honest I think we have enough static helper methods in the test base class, can we simply check the name of the field in the fuzzy test that needs this?